crates/extensions/tedge_health_ext/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+mod actor;
+use actor::HealthMonitorActor;
+use actor::HealthMonitorMessageBox;
+use tedge_actors::futures::channel::mpsc;
+use tedge_actors::Builder;
+use tedge_actors::CombinedReceiver;
+use tedge_actors::DynSender;
+use tedge_actors::LinkError;
+use tedge_actors::MessageSink;
+use tedge_actors::MessageSource;
+use tedge_actors::NoConfig;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::RuntimeRequestSink;
+use tedge_actors::ServiceConsumer;
+use tedge_mqtt_ext::MqttActorBuilder;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::TopicFilter;
+
+type HealthInputMessage = MqttMessage;
+type HealthOutputMessage = MqttMessage;
+
+pub struct HealthMonitorBuilder {
+    input_receiver: CombinedReceiver<HealthInputMessage>,
+    input_sender: mpsc::Sender<HealthInputMessage>,
+    output_sender: Option<DynSender<HealthOutputMessage>>,
+    signal_sender: mpsc::Sender<RuntimeRequest>,
+    name: String,
+}
+
+impl HealthMonitorBuilder {
+    pub fn new(name: String) -> Self {
+        let (input_sender, events_receiver) = mpsc::channel(10);
+        let (signal_sender, signal_receiver) = mpsc::channel(10);
+        let input_receiver = CombinedReceiver::new(events_receiver, signal_receiver);
+
+        Self {
+            input_receiver,
+            input_sender,
+            output_sender: None,
+            signal_sender,
+            name,
+        }
+    }
+
+    /// Connect this config manager instance to some mqtt connection provider
+    pub fn with_mqtt_connection(&mut self, mqtt: &mut MqttActorBuilder) -> Result<(), LinkError> {
+        let subscriptions = vec![
+            "tedge/health-check",
+            "tedge/health-check/c8y-device-management",
+        ]
+        .try_into()?;
+        //Register peers symmetrically here
+        mqtt.register_peer(subscriptions, self.input_sender.clone().into());
+        self.register_peer(NoConfig, mqtt.get_sender());
+        Ok(())
+    }
+}
+
+impl MessageSource<MqttMessage, NoConfig> for HealthMonitorBuilder {
+    fn register_peer(&mut self, _config: NoConfig, sender: DynSender<MqttMessage>) {
+        self.output_sender = Some(sender);
+    }
+}
+
+impl MessageSink<MqttMessage> for HealthMonitorBuilder {
+    fn get_sender(&self) -> DynSender<MqttMessage> {
+        self.input_sender.clone().into()
+    }
+}
```

Comment:
This actor is not a source nor a sink of `MqttMessage`. Technically, it sends and receives MQTT messages, yes, but these messages are of no use for another actor.

@albinsuresh This is why I'm in favor to remove these two traits from the actor API. Sending/receiving MQTT messages doesn't mean being a source/sink of MQTT messages, but implementing these two traits will let the compiler happy with `c8y_config_manager.set_mqtt_connection(health_check_actor)` (not fully correct as there is a clash with the `Config` type).

```suggestion
```

Reply:
Since we've agreed to go full-on with the `ServiceProvider`/`ServiceConsumer` abstractions, these two traits are definitely redundant. So I'm okay with removing them from the `tedge_actor_api` as well, as long as we have some solution for `FsWatchActor` and the likes.

Reply:
> Since we've agreed to go full-on with the `ServiceProvider`/`ServiceConsumer` abstractions, these two traits are definitely redundant. So I'm okay with removing them from the `tedge_actor_api` as well, as long as we have some solution for `FsWatchActor` and the likes.

I will push a PR in that direction then.